jvm/ml4ir-inference/pom.xml

```diff
@@ -10,6 +10,71 @@
   <artifactId>ml4ir-inference</artifactId>
   <packaging>jar</packaging>
 
+  <profiles>
+    <profile>
+      <activation>
+        <activeByDefault>false</activeByDefault>
+      </activation>
+
+      <id>integration_tests</id>
+
+      <properties>
+        <bundleLocation>${basedir}/../../python/</bundleLocation>
+        <runName>end_to_end_test</runName>
+      </properties>
+
+      <build>
+        <plugins>
+
+          <plugin>
+            <artifactId>exec-maven-plugin</artifactId>
+            <groupId>org.codehaus.mojo</groupId>
+            <executions>
+              <execution><!-- Run our version calculation script -->
+                <id>generate Python models</id>
+                <phase>pre-integration-test</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>${basedir}/scripts/generate_model.sh</executable>
+                  <arguments>
+                    <argument>${runName}</argument>
+                  </arguments>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-failsafe-plugin</artifactId>
```

Comment:
Note: as stated in the commit message, I had to move to failsafe for integration-test.

The tweak with the surefire was not working fine apparently, as it seems the test aren't run at the integration-test phase.

I hope it won't make a difference during the 3PP approval process?

Reply:
3PP should be fine